opendut-util/src/logging/mod.rs

```diff
@@ -229,6 +247,51 @@ pub enum OpenTelemetryConfig {
     #[default]
     Disabled,
 }
+
+#[derive(Clone)]
+struct MyArcMutex<T>(Arc<Mutex<T>>);
+
+impl Interceptor for MyArcMutex<Option<ConfidentialClientRef>> {
+    fn call(&mut self, mut request: Request<()>) -> Result<Request<()>, Status> {
+
+        let cloned_arc_mutex = Arc::clone(&self.0);
+
+        let mut token = None;
+        const MAX_RETRIES: i32 = 5;
+        
+        for _retries_left in (0..MAX_RETRIES).rev() {
+            match cloned_arc_mutex.try_lock() {
+                Ok(mutex_guard) => {
+                    let confidential_client= mutex_guard.clone();
+                    token = confidential_client.map(|client| client.get_token());
+                    break;
+                }
+                Err(error) => {
+                    thread::sleep(Duration::from_millis(500));
+                    eprintln!("Unable to acquire lock at the moment. Cause: {:?}", error);
+                }
+            };
+        }
+
+        let mut metadata_map = MetadataMap::new();
+        return match token {
+            None => { Ok(request) }
+            Some(token_result) => {
+                match token_result {
+                    Ok(token) => {
+                        let token_string = token.value.as_str();
+                        let bearer_header = format!("Bearer {token_string}");
+                        metadata_map.insert("authorization", MetadataValue::from_str(&bearer_header).expect("Cannot create metadata value from bearer header"));
+                        request.metadata_mut().insert("authorization", MetadataValue::from_str(&bearer_header).expect("Cannot create metadata value from bearer header"));
```

Comment:
Use the authorization header constant from http crate:
https://docs.rs/http/latest/http/header/index.html
Instead of the string "authorization"